### PR TITLE
feat(fts): engine integration

### LINKED ISSUE #8694: FTS: Integrate into the Prisma Client
Bump the engine once FTS is avialable.

### LINKED ISSUE #8695: FTS: Write integration tests
Need a couple end-to-end tests to verify it's working as expected.

### PR DESCRIPTION
**Pull request by millsp**
:warning: READ BEFORE MERGING

JSON filtering issues have been fixed, but one test (another) is now failing:

```ts
● client engine › expects(library) \| PRISMA_CLIENT_ENGINE_TYPE=undefined \| engineType=library
--
  | [2021-08-23T20:11:56Z]
  | [2021-08-23T20:11:56Z]     expect(received).toMatch(expected)
  | [2021-08-23T20:11:56Z]
  | [2021-08-23T20:11:56Z]     Expected substring: "library"
  | [2021-08-23T20:11:56Z]     Received string:    "binary"
  | [2021-08-23T20:11:56Z]
  | [2021-08-23T20:11:56Z]       109 \|             `)
  | [2021-08-23T20:11:56Z]       110 \|
  | [2021-08-23T20:11:56Z]     > 111 \|           expect(prisma._clientEngineType).toMatch(expectedClientEngine)
  | [2021-08-23T20:11:56Z]           \|                                            ^
  | [2021-08-23T20:11:56Z]       112 \|           await prisma.$disconnect()
  | [2021-08-23T20:11:56Z]       113 \|         })
  | [2021-08-23T20:11:56Z]       114 \|       }
  | [2021-08-23T20:11:56Z]
  | [2021-08-23T20:11:56Z]       at Object.<anonymous> (src/__tests__/integration/happy/client-engine/test.ts:111:44)
  | [2021-08-23T20:11:56Z]           at runMicrotasks (<anonymous>)
```

This is happening since this [Prisma Engine release](https://github.com/prisma/prisma/pull/8844) but was missed in the logs while we were fixing another issue at the same time.

We can get `integration/happy/client-engine` to work with [the version just before](https://github.com/prisma/prisma/pull/8829). However, that version was also the one we were waiting fixes for (tests failing on JSON filtering).

TL;DR; JSON filtering was fixed on the latest engines but something around the `engineType` parsing seems to have changed. Was this an intended change, does it need integration?

The `client-engine` test has been disabled temporarily for BuildKite to pass (if we want to release like this), but this will still fail later on in `e2e-tests`. If we provide a fix in time, we need to re-enable this skipped test.

Fixes https://github.com/prisma/prisma/issues/8695
Fixes https://github.com/prisma/prisma/issues/8694

### CODE REVIEW

File: packages/client/src/__tests__/integration/happy/full-text-search/test.ts [hunk 1]
```diff
@@ -0,0 +1,186 @@
+import path from 'path'
+import { generateTestClient } from '../../../../utils/getTestClient'
+import { tearDownPostgres } from '../../../../utils/setupPostgres'
+import { migrateDb } from '../../__helpers__/migrateDb'
+
+// @ts-ignore trick to get typings at dev time
+import type { PrismaClient } from './node_modules/@prisma/client'
```
Comment (Jolg42) [inline packages/client/src/__tests__/integration/happy/full-text-search/test.ts:7]: We should do that in all our tests maybe! 👍🏼

**Comment by pimeys** [comment]
Yup, it's an internal api change in the datamodel parser that is leaking to a public api. Fixing this now, or finding a good fix.

**Comment by Jolg42** [comment]
Engine PR is https://github.com/prisma/prisma-engines/pull/2175

**Comment by pimeys** [comment]
merged.

File: packages/client/src/__tests__/integration/happy/client-engine/test.ts [hunk 2]
```diff
@@ -73,7 +73,7 @@ function buildTests() {
           envVar,
           value,
         )
-        test(`expects(${expectedClientEngine}) | ${envVar}=${value} | engineType=${engineType}`, async () => {
+        test.skip(`expects(${expectedClientEngine}) | ${envVar}=${value} | engineType=${engineType}`, async () => {
```
Comment (Jolg42) [inline packages/client/src/__tests__/integration/happy/client-engine/test.ts:76]:
```suggestion
        test(`expects(${expectedClientEngine}) | ${envVar}=${value} | engineType=${engineType}`, async () => {
```

File: packages/client/src/__tests__/integration/happy/full-text-search/test.ts [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        name: '0 2 4 6 8',
+      },
+    })
+
+    await prisma.user.create({
+      data: {
+        email: 'email3@email.io',
+        name: '1 3 5 7 9',
+      },
+    })
```
Comment (Jolg42) [inline packages/client/src/__tests__/integration/happy/full-text-search/test.ts:43]: Promise.all would be faster I think

**Comment by Jolg42** [comment]
Looks like current tests only cover the "happy path" when all of this is true:
- provider is postgresql (is there an error when using another provider?)
- search with a valid operator (|,!,&,<->) (what happens if the operator is invalid?)
- search returns results (returns empty array if no result?)

**Review by Jolg42 — APPROVED** [review]
Thanks @millsp for adding more tests!
